src/Fixer/Phpdoc/PhpdocArrayStyleFixer.php

```diff
@@ -0,0 +1,160 @@
+     * Must run after AlignMultilineCommentFixer, CommentToPhpdocFixer, PhpdocIndentFixer, PhpdocScalarFixer, PhpdocToCommentFixer, PhpdocTypesFixer.
+     */
+    public function getPriority(): int
+    {
+        return 1;
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('strategy', 'Which part of the conversion - brackets (`[]`) to `array` to `list` - to perform.'))
+                ->setAllowedValues([self::STRATEGY_FROM_ARRAY_TO_LIST, self::STRATEGY_FROM_BRACKETS_TO_ARRAY, self::STRATEGY_FROM_BRACKETS_TO_LIST])
+                ->setDefault(self::STRATEGY_FROM_BRACKETS_TO_ARRAY)
+                ->getOption(),
+        ]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        for ($index = $tokens->count() - 1; $index > 0; --$index) {
+            if (!$tokens[$index]->isGivenKind([T_DOC_COMMENT])) {
+                continue;
+            }
+
+            $docBlock = new DocBlock($tokens[$index]->getContent());
+
+            foreach ($docBlock->getAnnotations() as $annotation) {
+                if (!$annotation->supportTypes()) {
+                    continue;
+                }
+
+                $typeExpression = $annotation->getTypeExpression();
+                if (null === $typeExpression) {
+                    continue;
+                }
+
+                $annotation->setTypes([$this->normalize($typeExpression->toString())]);
+            }
+
+            $newContent = $docBlock->getContent();
+            if ($newContent === $tokens[$index]->getContent()) {
+                continue;
+            }
+
+            $tokens[$index] = new Token([T_DOC_COMMENT, $newContent]);
+        }
+    }
+
+    protected function normalize(string $type): string
+    {
+        $typeExpression = new TypeExpression($type, null, []);
+
+        $typeExpression->walkTypes(function (TypeExpression $type): void {
+            if ($type->isUnionType()) {
+                return;
+            }
+
+            $value = $type->toString();
+
+            if (Preg::match('/^\??\s*[\'"]/', $value)) {
+                return;
+            }
+
+            if (str_starts_with($value, '?')) {
+                $value = '?'.$this->fixType(substr($value, 1));
+            } else {
+                $value = $this->fixType($value);
+            }
+
+            \Closure::bind(static function () use ($type, $value): void {
+                $type->value = $value;
+            }, null, TypeExpression::class)();
+        });
+
+        return $typeExpression->toString();
+    }
+
+    private function fixType(string $type): string
+    {
+        if (self::STRATEGY_FROM_ARRAY_TO_LIST !== $this->configuration['strategy']) {
+            do {
+                $type = Preg::replace('/(.+)\[\]/', 'array<$1>', $type, -1, $count);
```

Comment:
will this work with type like `array{'a', 'b[]', 'c'}` or `\Closure(a, b[]): void`?